pkg/dispatcher/dispatcher.go

```
@@ -1,41 +1,80 @@
 package dispatcher
 
-import "context"
+import (
+	"context"
+	"github.com/justtrackio/gosoline/pkg/appctx"
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/log"
+	"sync"
+)
 
-type Result map[string]error
-type Callback func(ctx context.Context, event interface{}) error
+type (
+	dispatcherCtxKey string
+
+	Callback func(ctx context.Context, event interface{}) error
+	Result   map[string]error
+)
 
 type Dispatcher interface {
 	Fire(ctx context.Context, name string, event interface{}) Result
+	On(name string, call Callback)
+	Reset()
 }
 
-type dispatcher struct{}
+type dispatcher struct {
+	logger    log.Logger
+	mx        sync.RWMutex
+	listeners map[string][]Callback
+}
 
-var listeners = map[string][]Callback{}
-var d = &dispatcher{}
+func ProvideDispatcher(ctx context.Context, config cfg.Config, logger log.Logger) Dispatcher {
+	dispatcher, _ := appctx.Provide(ctx, dispatcherCtxKey("DispatcherRepository"), func() (Dispatcher, error) {
```

We shouldn't ignore errors and instead handle them in a proper way.